### PR TITLE
Add Response.unauthorized (401) constructor

### PR DESCRIPTION
**Pull request by develogo**

### CODE REVIEW

**Review by kevmoo — CHANGES_REQUESTED** [review]
Cool! Please add a changelog entry and bump the version to `1.4.0` – since this is adding new API!

File: pkgs/shelf/lib/src/response.dart [hunk 1]
```diff
@@ -180,6 +180,24 @@ class Response extends Message {
           encoding: encoding,
         );
 
+  /// Constructs a 401 Unauthorized response.
+  ///
+  /// This indicates that the request is unauthorized due to invalid
```
Comment (kevmoo) [inline pkgs/shelf/lib/src/response.dart:185]: invalid...what? Might want to fix this sentence, too!

File: pkgs/shelf/test/response_test.dart [hunk 2]
```diff
@@ -106,6 +106,13 @@ void main() {
     });
   });
 
+  group('Response.unauthorized:', () {
+    test('sets body', () {
+      var response = Response.unauthorized('Unauthorized');
+      expect(response.readAsString(), completion(equals('Unauthorized')));
```
Comment (kevmoo) [inline pkgs/shelf/test/response_test.dart:112]: A tiny bit redundant, but you should probably check the status code here, too!

**Comment by kevmoo** [comment]
@develogo – please also update the pubspec version to match the changelog

**Comment by develogo** [comment]
@kevmoo updated, sorry for the lack of attention

**Comment by kevmoo** [comment]
> @kevmoo updated, sorry for the lack of attention

You're learning! No worries. Thanks for the contribution...

**Review by kevmoo — APPROVED** [review]

**Pull request merged**